stable/datadog/templates/container-agent.yaml

```diff
@@ -0,0 +1,161 @@
+    {{- end }}
+    {{- if .Values.datadog.hostname }}
+    - name: DD_HOSTNAME
+      value: {{ .Values.datadog.hostname | quote }}
+    {{- end }}
+    {{- if .Values.datadog.acInclude }}
+    - name: DD_AC_INCLUDE
+      value: {{ .Values.datadog.acInclude | quote }}
+    {{- end }}
+    {{- if .Values.datadog.acExclude }}
+    - name: DD_AC_EXCLUDE
+      value: {{ .Values.datadog.acExclude | quote }}
+    {{- end }}
+    {{- if not .Values.clusterAgent.enabled }}
+    {{- if .Values.datadog.leaderElection }}
+    - name: DD_LEADER_ELECTION
+      value: {{ .Values.datadog.leaderElection | quote}}
+    {{- end }}
+    {{- if .Values.datadog.leaderLeaseDuration }}
+    - name: DD_LEADER_LEASE_DURATION
+      value: {{ .Values.datadog.leaderLeaseDuration | quote }}
+    {{- end }}
+    {{- if .Values.datadog.collectEvents }}
+    - name: DD_COLLECT_KUBERNETES_EVENTS
+      value: {{.Values.datadog.collectEvents | quote}}
+    {{- end }}
+    {{- else }}
+    - name: DD_CLUSTER_AGENT_ENABLED
+      value: {{ .Values.clusterAgent.enabled | quote }}
+    - name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
+      value: {{ template "datadog.fullname" . }}-cluster-agent
+    - name: DD_CLUSTER_AGENT_AUTH_TOKEN
+      valueFrom:
+        secretKeyRef:
+          name: {{ template "datadog.fullname" . }}-cluster-agent
+          key: token
+    {{- end }}
+    {{- if .Values.datadog.podLabelsAsTags }}
+    - name: DD_KUBERNETES_POD_LABELS_AS_TAGS
+      value: '{{ toJson .Values.datadog.podLabelsAsTags }}'
+    {{- end }}
+    {{- if .Values.datadog.podAnnotationsAsTags }}
+    - name: DD_KUBERNETES_POD_ANNOTATIONS_AS_TAGS
+      value: '{{ toJson .Values.datadog.podAnnotationsAsTags }}'
+    {{- end }}
+    {{- if semverCompare "^1.7-0" .Capabilities.KubeVersion.GitVersion }}
+    - name: DD_KUBERNETES_KUBELET_HOST
+      valueFrom:
+        fieldRef:
+          fieldPath: status.hostIP
+    {{- end }}
+    {{- if .Values.datadog.logsEnabled }}
+    - name: DD_LOGS_ENABLED
+      value: {{.Values.datadog.logsEnabled | quote}}
+    {{- end }}
+    {{- if .Values.datadog.logsConfigContainerCollectAll }}
+    - name: DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL
+      value: {{.Values.datadog.logsConfigContainerCollectAll | quote}}
+    {{- end }}
+    {{- if .Values.datadog.criSocketPath }}
+    - name: DD_CRI_SOCKET_PATH
+      value: {{ .Values.datadog.criSocketPath | quote }}
+    {{- end }}
+    {{- if not .Values.datadog.livenessProbe }}
+    - name: DD_HEALTH_PORT
+      value: "5555"
+    {{- end }}
+    {{- if .Values.datadog.useDogStatsDSocketVolume }}
+    - name: DD_DOGSTATSD_SOCKET
+      value: "/var/run/datadog/dsd.socket"
+    {{- end }}
+    {{- if and .Values.clusterAgent.clusterChecks.enabled  (not .Values.clusterchecksDeployment.enabled) }}
+    - name: DD_EXTRA_CONFIG_PROVIDERS
+      value: "clusterchecks"
+    {{- end }}
+{{- if .Values.datadog.env }}
+{{ toYaml .Values.datadog.env | indent 4 }}
```

Comment:
we should probably document this section to mention that if used jointly with separate containers, this will lead to injecting this env in every container. Or we should have a separate `.Values.<container>.env` section per container. What do you think is best?

Reply:
These env values are only injected in the agent container so far, not in other containers. But I like the idea to also have a separate section for the agent container in the daemonset. Should we also provide a way to set env common to several containers (could be useful for instance to set the log level and that kind of stuff) ?

Reply:
So in the end for now we are able to set per container env with `daemonset.containers.<container>.env`.